### ISSUE TITLE
More colour variants?

### ISSUE THREAD

**Issue by AcarBurak** [tsujan/Kvantum#188]
Thanks for this very elegant utility.

Kvantum has many configuration options (seemingly especially for theme creators), but being unable to change button colours, lineedits etc. the ordinary users like me are obliged to use the pre-sets you provide with Kvantum (unless they're happy with their custom and broken colour schemes). 

As for the pre-sets included with Kvantum, though they're fine in themselves they're too similar. Hence my humble colour combination suggestions below. Please, ignore any or all of them if it's a big trouble to implement or if you find them ugly or unnecessary; rejection won't make me upset as I'm obviously no artist anyway. (All qtcurve styles customised by qt5ct)
#5c8a8a    #283c3c
![5c8a8a 283c3c](https://user-images.githubusercontent.com/1426842/37864651-04533e5a-2f83-11e8-94d6-06df61d0b817.png)
#5e7d6d    #27342d
![5e7d6d 27342d](https://user-images.githubusercontent.com/1426842/37864662-321160ec-2f83-11e8-837d-4bca663eb24f.png)
#9a8678    #425a5c
![9a8678 425a5c](https://user-images.githubusercontent.com/1426842/37864663-3232bbde-2f83-11e8-97da-2521d9698ed2.png)
#9bbec1    #9bbec1
![9bbec1 9bbec1](https://user-images.githubusercontent.com/1426842/37864664-32532af4-2f83-11e8-81b8-34e6f3841033.png)
#425a5c    #192223
![425a5c 192223](https://user-images.githubusercontent.com/1426842/37864665-3274905e-2f83-11e8-9962-d9ee63f4fce0.png)
#523b35    #291610
![523b35 291610](https://user-images.githubusercontent.com/1426842/37864666-32a84520-2f83-11e8-8540-5fc8fcf0eabf.png)
#647d86    #2d383c
![647d86 2d383c](https://user-images.githubusercontent.com/1426842/37864667-32c9bda4-2f83-11e8-8226-6d5f0b0deba1.png)
#684b44    #425a5c
![684b44 425a5c](https://user-images.githubusercontent.com/1426842/37864668-32fda4a2-2f83-11e8-988f-85d21d58704a.png)
#64726e    #293231
![64726e 293231](https://user-images.githubusercontent.com/1426842/37864669-33214a60-2f83-11e8-9db2-22b0395a9657.png)
#87826c    #403b2f
![87826c 403b2f](https://user-images.githubusercontent.com/1426842/37864671-338815a6-2f83-11e8-9512-742ce0850208.png)
#136653    #192223
![136653 192223](https://user-images.githubusercontent.com/1426842/37864672-33ab606a-2f83-11e8-8483-cfc3914f484c.png)
#577779    #212e2f
![577779 212e2f](https://user-images.githubusercontent.com/1426842/37864673-33cc4ffa-2f83-11e8-9f45-250f7afca66c.png)
#993937    #202d3a
![993937 202d3a](https://user-images.githubusercontent.com/1426842/37864674-34024330-2f83-11e8-9dc1-f2705db2687c.png)
#a29b81    #c7c3b3
![a29b81 c7c3b3](https://user-images.githubusercontent.com/1426842/37864675-34367d76-2f83-11e8-8cad-0aeff6982ffb.png)

**Comment by tsujan** [comment]
@AcarBurak  Kvantum's code is full of "troubles" and I don't mind to add more ;)

However, there's a big misunderstanding about "color schemes". Please first take a look at my comments on this page -- it includes useful info: https://github.com/tsujan/Kvantum/issues/186

Style engines like Oxygen, Breeze and QtCurve -- actually, all style engines before Kvantum -- made the concept of "color scheme" because they drew widgets by using them.

On the one hand, I wasn't satisfied with the limitations color schemes imposed on how widgets looked. A glance at your attached screenshots show that all of them have the same look and feel -- just the colors are changed. That was what could be done before Kvantum. On the other hand, the SVG support of Qt was excellent (I was inspired by KDE4's Plasma). So, at first, I was happy that "QuantumStyle" had been made but, soon, I found that it was too elementary. Anyhow, I wanted virtually no limit to how widgets could be drawn. Fortunately, `QStyle` was flexible enough to provide what I wanted. Hence, Kvantum was started.

Kvantum draws widgets by using SVG elements, which could have complex layers of gradients and don't follow any scheme. Kvantum also uses a mixture of Qt's standard text colors and its own complex text colors to draw labels but they should match SVG elements. Also, in rare cases, Kvantum uses Qt's standard background colors. Therefore, in short, Kvantum has gone *far beyond* the concept of "color scheme".

For having different colors for a Kvantum theme, a totally new theme should be made -- and I've done it here and there, as far as the new look has been acceptable to me. It takes me a day to make a new Kvantum theme and the day after that, I polish it. Sometimes, polishing will continue indefinitely. As I've explained in the Kvantum doc, making new themes for Kvantum isn't a piece of cake -- it needs SVG skills and patience -- but that's what you pay to have control over virtually all aspects of widgets.

To summarize: "Color scheme" isn't a feature for Kvantum; it was an obstacle removed a long time ago.

**Comment by tsujan** [comment]
As I like to explain things with examples -- and although all Kvantum themes could serve as examples -- please install the latest git Kvantum, activate `KvFlatLight` with Kvantum Manager and see that no color scheme is enough for it it, although it's a flat and simple theme.

**Comment by AcarBurak** [comment]
tsujan, sorry for not being clear enough and forcing you to explain all this. I know all the above (I guess, except for the paragraph before the last) and because of that I've asked you to include more "colour schemes" meaning more themes, that is, in addition to KvBeige etc. for instance, KvHoki, KvCuttySark etc having such colours like above.

To explain further, now, I like KvBeige (especially its more easily noticeable insets or sunken appearances) but I'd like to have some other colour combinations as well beside the default beige and white-like one, the colours as I've shown above. I can't do it myself as you've explained; I can change some colours, but not the colours of the buttons. That's why I've asked for more schemes, meaning more pre-set themes. But since it takes so long time and energy, I give up. Please feel free to close this.

**Comment by tsujan** [comment]
Oh, sorry for my irrelevant explanations but they could serve as a reference in future.

So, you like the *form* of `KvBeige` but want it with a different color? OK! `KvBeige` is my favorite light theme, especially because of its tabs. I activate it once in a while.

Since my time is limited -- now I'm working on the checkbox sizes on menus -- I'll be glad to make *one* color variant for `KvBeige`. Please tell me about a color of your choice. We'll discuss about the details as well as our different tastes later, when the work is started.

Also, I'll be happy to make color variants of some other themes if you want.

**Comment by tsujan** [comment]
> I can't ask (and nor want) you to create anything you don't like. 

I've made themes I didn't like but Kvantum users wanted them. It's enough for me not to hate them ;) Just joking.... Sometimes, I use `KvAdapta` for a day or two, although I didn't like it at first.

I'll experiment with the colors you mentioned. It might take a while because, after doing other jobs, I should change several gradients to see the real look. I'll tell you, with screenshots, when the work is started.

**Comment by tsujan** [comment]
A preliminary discussion:

To me, something near `#9a8678` (or a little darker, like `#8c7a6d`) looks like a decent dark background color that won't be boring after a while. `#425a5c` and `#647d86` could be too poignant with constant use -- I made `KvCyan` but can rarely use it ;) IMHO, background colors are better to be "neutral". Brown, beige, light gray, dark gray, dark green, light blue,... are "neutral" because they can be found in nature's vast spaces.

What do you think?

**Comment by AcarBurak** [comment]
Very fine, but a bit hurt by #425a5c comment, "poignant"! And this is despite Kvantum having the most poignant of it all as default, #3d3d3e!

All joking off course, it's up to you.

On the other hand, I can try one of the dark pre-sets (Default for instance) with #425a5c; maybe as a window colour it plays nice with the pre-set button colours.

**Comment by AcarBurak** [comment]
And #8c7a6d seems better than #9a8678.

**Comment by AcarBurak** [comment]
#9a8678 is very fine to me.

**Comment by tsujan** [comment]
OK. I think `#9a8678` has also enough space for making the sunken parts. I suggest the fancy name "KvAlmond" for the theme.

**Comment by AcarBurak** [comment]
Perfect.

**Comment by tsujan** [comment]
Will add `KvAlmond` to the collection :) Probable suggestions about variants of other themes would also be welcome. Oh, ideas about totally new themes too.

**Comment by AcarBurak** [comment]
Sorry, I've meant the darker, 8c7a6d, but if you prefer 9a8678, that's as fine too.

**Comment by tsujan** [comment]
Both are good to me; will experiment with `#8c7a6d` too.

**Comment by AcarBurak** [comment]
A non-poignant example with natural colours (sea, sky) just for inspiration:
![blue](https://user-images.githubusercontent.com/1426842/37873491-8bf80f5e-3026-11e8-8d1f-e995dd10c0de.png)

**Comment by tsujan** [comment]
IMO, that's a very nice color but with flat themes -- the sky/sea looks 2-dimensional.

After playing with several colors/gradients, I found this combination almost serene:

![1](https://user-images.githubusercontent.com/981076/37874274-556e6f26-3041-11e8-8626-4721bb8a1920.png)

I made the background color a little darker because it looked too light when one's eyes got accustomed to darkness (I'd seen it only against white backgrounds before making the theme).

However, it still doesn't seem completely fine to me; maybe the background isn't dark enough or, perhaps, some contours are too dark. I should find what's wrong and fix it.

What do you think?

**Comment by AcarBurak** [comment]
Looks fine; what about the insets:
![insets](https://user-images.githubusercontent.com/1426842/37874364-3fc5c06c-3036-11e8-8942-434ae13e9337.png)

**Comment by tsujan** [comment]
It' almost there but something tickles my senses:

![2](https://user-images.githubusercontent.com/981076/37874411-efca0286-3043-11e8-96f5-1d6b1e3ff929.png)

Maybe, line-edits/spinboxes should be made lighter; or the toggled color has too much of red;... I'll find it.

**Comment by AcarBurak** [comment]
It looks almost fabulous. I say "almost" because though very fine, those insets look best with light colours I guess as in the default KvBeige, but there's nothing to do about it seemingly. All the same very fine.

By the way, background colour is fine to me as well; I would experiment with the lighter instead of darken version. And could you show sliders as well? And how about making the selected button in the toolbar with inset as well?

**Comment by tsujan** [comment]
> those insets look best with light colours

You're right. Making deep insets with dark colors is a bit harder. I'll work on them.

As for toolbar buttons, since the toolbar is sunken, a sunken toggled button would look ugly but the pressed buttons are all sunken and darker.

Please note that you're seeing the screenshots on a white background here. If all apps had this theme, you'd see that it's a very light dark theme ;)

**Comment by AcarBurak** [comment]
Agree, with all.

**Comment by AcarBurak** [comment]
Please consider making sliders more (or much more) prominent as I can't see them in some (many) other pre-sets (as in the default Fusion style).

**Comment by tsujan** [comment]
You mean the slider and not the scroll-bar slider? The latter isn't visible in the above screenshots because it's transient. Yes, it should be more visible. The above screenshots are only about the general look.

**Comment by AcarBurak** [comment]
Yes, the scroll-bar slider. … Though that progress colour isn't very easily noticeable either.

**Comment by tsujan** [comment]
> Though that progress colour isn't very easily noticeable either.

I know. And I thought I was the only one who's obsessive about such things :D

**Comment by AcarBurak** [comment]
:)

**Comment by tsujan** [comment]
I worked on contours and some gradients. A few details aside (checkboxes included), I think it's stylish now:

![3](https://user-images.githubusercontent.com/981076/37875840-d6117340-3059-11e8-9281-a2b5fa82edfb.png)

I take a distance from it for a while and then, will return to it to see if there's a problem.

**Comment by AcarBurak** [comment]
Yes, very elegant.

**Comment by tsujan** [comment]
I came back to this and saw that the bg color was a bit hot. After I reduced its saturation by 4 units and did the same thing with most gradients, it became better. This is pcmanfm-qt with the theme and the Breath icon set:

![pcmanfm-qt](https://user-images.githubusercontent.com/981076/37917655-34205084-3134-11e8-98a7-a05a8b3a4f1e.png)

I still feel something's wrong.Some apps look elegant with it, some not... However, I may have lost my objectivity about it. What do you think?

**Comment by AcarBurak** [comment]
I may only (weakly) say that the white on the dark brown buttons of toolbar on the upper left seems a bit bright. Those bright green folders seem a bit out of place to my eyes, maybe they upset you regardless of the KvFabulous!

**Comment by tsujan** [comment]
I think it isn't dark enough. Actually, most icon themes look pale on the side-pane because of that. Please compare the above screenshot with the following one without paying attention to icons or the toggled button gradient and tell me your opinion:

![pcmanfm-qt1](https://user-images.githubusercontent.com/981076/37920699-da3d8e98-313c-11e8-8e71-d584767e25ea.png)

Sorry if I ask too many questions -- you've participated in the long/slow process I have while making a new theme that isn't a clone of gtk themes ;)

**Comment by AcarBurak** [comment]
I would be glad if I help in any way.

I would choose the first one, though the real thing on my desktop may seem a bit different when installed.

The second one seems to loose its chocolatey feeling and approaches to be a shade of black-brown mixture, a different territory.

**Comment by tsujan** [comment]
> though the real thing on my desktop may seem a bit different when installed.

You've seen them against the extremely white background of this page. I attached both themes to this comment -- hopefully correctly. Whenever you have enough time, please test them. If you really like the lighter one, I'll include both in Kvantum.

[themes.zip](https://github.com/tsujan/Kvantum/files/1848826/themes.zip)

**Comment by AcarBurak** [comment]
Dark seems a tad better.

**Comment by tsujan** [comment]
> Dark seems a tad better.

Its elements should be corrected; I just changed some gradients.

**Comment by AcarBurak** [comment]
Maybe irrelevant, but I can only state my personal taste: I generally find flat buttons plain and not very functional; other parts can be fine with either a flat style or some subtle movements for which svg is a good tool I feel.

And generally pastel colours, obviously not meaning non-lively; pastel, lively or not. Same for the gradients, generally, as well, pastel or even matte. Below is such a gradient example of Adwaita adapted and compiled by some sed and sassc commands from my (above mentioned) blue fusion colours:
![adwbtn](https://user-images.githubusercontent.com/1426842/37942890-fc8989b6-317d-11e8-80fe-e880e6bab8ac.png)

Other than that, not a moc-up, but some colour combinations, though pcmanfm-qt's fusion style buttons (gradients) especially the tab buttons don't look handsome.
![lblue](https://user-images.githubusercontent.com/1426842/37942930-1dabb11e-317e-11e8-9ccb-481279306546.png)
![lcream](https://user-images.githubusercontent.com/1426842/37942931-1dd078e6-317e-11e8-8250-5ac2a47c40df.png)
![lgreen](https://user-images.githubusercontent.com/1426842/37942932-1df3f348-317e-11e8-9338-64e47d54101f.png)
![lyellow](https://user-images.githubusercontent.com/1426842/37942933-1e29658c-317e-11e8-8aa0-446d42aac884.png)
![lyellow2](https://user-images.githubusercontent.com/1426842/37942934-1e6c1210-317e-11e8-9893-e30e7594bc6f.png)

**Comment by AcarBurak** [comment]
Forgot colours. Window and Base:
#9db6bd    #c9bea3
#c9bea3    #a5b6b3
#9ab2ae    #d1ccbc
#ddcc97    #a5b6b3
#ddc483    #94bbbb

**Comment by tsujan** [comment]
Good colors but, currently, I can't imagine an elegant theme with them. I guess they may look fine in flat themes like `KvFlatLight`. An experiment for future.

Believe it or not, I'm still modifying the brown theme ;)

**Comment by tsujan** [comment]
I got satisfied with the look of KvBrown after modifying it a thousand times, started to use it happily, and made a commit.

In the case of non-flat themes, finding the "correct" gradients is hard with some bg colors and practically impossible with some others. Of course, that will be the case only if a nice theme is expected; otherwise, any color could be used with Kvantum. And "nice" is a very relative and subjective concept.

I close this but may experiment with *flat* colorful themes later.
